types/web3/types.d.ts

```diff
@@ -85,5 +85,10 @@ export interface Subscribe<T> {
     on(type: "error", handler: (data: Error) => void): void;
 }
 
-export class Shh {} // TODO: Type
+export interface Shh {
+    generateSymKeyFromPassword(
+        password: string,
+        callback?: Callback<string>
+    ): Promise<string>;
```

Comment:
I guess the [docs](https://web3js.readthedocs.io/en/1.0/web3-shh.html?highlight=generateSymKeyFromPassword#id61) are incorrect here and this functions returns a promise for a string, not a string. Question: Does it also return a promise when a callback is provided? I would say an overload would be more appropriate here. I created an [issue](https://github.com/ethereum/web3.js/issues/1924). Even if a promise is returned when a `callback` parameter is given, we should not type this bad behaviour here. So I would say we make two versions of the this function. One which takes just a string and returns a `Promise<string>` and one which takes a string and a callback function and returns `void`.

Reply:
Fully agreed.